src/orion/core/worker/warm_start/multi_task_wrapper.py

```diff
@@ -0,0 +1,249 @@
+            default_value=TARGET_TASK_ID,
+        )
+        new_space = copy.deepcopy(space)
+        new_space.register(task_label_dimension)
+        return new_space
+
+    def _is_compatible(self, trial_from_other_experiment: Trial) -> bool:
+        """Used to check if a trial is compatible with the current experiment and can be reused.
+
+        The trial will have a different task ID than the current task.
+
+        NOTE: this assumes that the trial from the other experiment comes from the knowledge base,
+        and that it therefore doesn't already have a task ID.
+        """
+        return trial_from_other_experiment in self.space
+
+    def warm_start(
+        self, warm_start_trials: list[tuple[ExperimentConfig, list[Trial]]]
+    ) -> None:
+        """Observe some of the given trials to warm-start the HPO algorithm.
+
+        These experiments and their trials were fetched from some storage, and
+        are believed to be somewhat similar to the current on-going experiment.
+        By default, this wrapper only considers the trials which are compatible
+        with the current space.
+
+        Parameters
+        ----------
+        warm_start_trials:
+            Dictionary mapping from ExperimentConfig objects (containing the experiment config) to
+            the list of Trials associated with that experiment.
+        """
+        logger.info(
+            "Warm-starting using only multi-task training with compatible trials, since the algo "
+            "isn't warm-starteable."
+        )
+
+        # Dict mapping from task ID to list of compatible trials.
+        compatible_trials: dict[int, list[Trial]] = defaultdict(list)
+        task_ids_to_experiments: dict[int, ExperimentConfig] = {}
+        for i, (experiment_info, trials) in enumerate(warm_start_trials):
+            # Start the task ids at 1, so the current experiment has task id 0.
+            task_id = i + 1
+            for trial in trials:
+                # Drop the point if it doesn't fit inside the current space.
+                # IDEA: Possibly add some 'translation' logic in an eventual Knowledge Base
+                # implementation.
+                if self._is_compatible(trial):
+                    compatible_trials[task_id].append(trial)
+                    task_ids_to_experiments[task_id] = experiment_info
+
+            n_compatible_trials = len(compatible_trials[task_id])
+            logger.info(
+                f"Experiment {experiment_info} has {len(trials)} trials in "
+                f"total, out of which {n_compatible_trials} were found to be "
+                f"compatible with the target experiment."
+            )
+
+        if not compatible_trials:
+            logger.info("No compatible trials found!")
+            return
+
+        # Note: If there are no compatible trials found, this would allow multiple attempts at
+        # warm-starting. This isn't used though.
+        if self._total_warm_start_trials:
```

Comment:
This should be tested.

Reply:
aff8c7b3d62ba5c46b1a245b80cf360497ef108b